esp-hal/src/spi/master.rs

```diff
@@ -2409,20 +2404,26 @@ pub trait InstanceDma: Instance {
     }
 }
 
-#[cfg(not(any(esp32, esp32s2)))]
-fn reset_dma_before_usr_cmd(reg_block: &RegisterBlock) {
-    reg_block.dma_conf().modify(|_, w| {
+fn reset_dma_before_usr_cmd(_reg_block: &RegisterBlock) {
+    #[cfg(not(any(esp32, esp32s2)))]
+    _reg_block.dma_conf().modify(|_, w| {
         w.rx_afifo_rst()
             .set_bit()
             .buf_afifo_rst()
             .set_bit()
             .dma_afifo_rst()
             .set_bit()
     });
-}
 
-#[cfg(any(esp32, esp32s2))]
-fn reset_dma_before_usr_cmd(_reg_block: &RegisterBlock) {}
+    // Wait for a few clock cycles for the DMA to fill the SPI async FIFO,
+    // before starting the SPI
+    let cycles = 20;
```

Comment:
This is... concerning. `1` makes tests pass (except on C2, which needs 2), but the example needs `10`. With `5` I can get 5 correct bytes, the rest incorrect. This almost makes sense, except I'd expect the DMA to fill the buffer faster than 1 byte per clock cycle.

The examples should be optimised with the same settings as the hil-tests.

Reply:
It might be better to swap the order back. Best to use those clock cycles to setup the RX side of the DMA rather than waste them sleeping.

I did spot a `SPI_DMA_OUTFIFO_EMPTY` register bit that you might be able to poll to see if the async buf has some content in it. Not all chips have it though

Reply:
Ah nvm, looks like you figured something out

Reply:
Yup, and I have absolutely no idea if it's a coincidence that it works, or it's the solution. Anyhow, it looks to be doing its thing.